### PR TITLE
Feature/create reservation

### PR DESCRIPTION
**Pull request by xandervedder**

### CODE REVIEW

File: src/main/java/nl/asd/workplace/domain/Building.java [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 
 public class Building {
     private BuildingId id;
     private String name;
-    private HashMap<Day, OpeningHours> openingHours;
+    // TODO: Don't think HashMap is the right choice here..., maybe a HashSet or some other Value object?
+    private HashMap<DayOfWeek, OpeningTime> openingHours;
```
Comment (Rubenvdbrink) [inline src/main/java/nl/asd/workplace/domain/Building.java:16]: Maybe simply use a list with a value object {int dayofweek, LocalTime from, LocalTime to}

Comment (xandervedder) [reply src/main/java/nl/asd/workplace/domain/Building.java:16]: I was a list before this one, but the size of that list isn't going to change (there are 7 days in a week), which is why I did a HashMap, but it still doesn't feel nice

File: src/main/java/nl/asd/reservation/domain/Reservation.java [hunk 2]
```diff
@@ -3,70 +3,86 @@
...
+        this.setWorkplace(workplace);
     }
 
     public ReservationId getId() {
         return id;
     }
 
     public void setId(ReservationId id) {
+        if (id == null) {
+            throw new IllegalArgumentException("Cannot create reservation without an id");
```
Comment (JortWillemsen) [inline src/main/java/nl/asd/reservation/domain/Reservation.java:35]: Change to: Reservation id cannot be null

File: src/main/java/nl/asd/reservation/domain/Reservation.java [hunk 3]
```diff
@@ -3,70 +3,86 @@
...
-    public void setCreatedAt(LocalDate createdAt) {
-        this.createdAt = createdAt;
+    public LocalDate getReservationDate() {
+        return reservationDate;
     }
 
-    public LocalDate getReservationDate() { return reservationDate; }
+    public void setReservationDate(LocalDate reservationDate) {
+        if (reservationDate == null) {
+            throw new IllegalArgumentException("Cannot create a reservation without a reservation date");
```
Comment (JortWillemsen) [inline src/main/java/nl/asd/reservation/domain/Reservation.java:51]: Change to: Reservation date cannot be null

File: src/main/java/nl/asd/reservation/domain/Reservation.java [hunk 4]
```diff
@@ -3,70 +3,86 @@
...
+        this.reservationDate = reservationDate;
+    }
 
     public ReservationType getType() {
         return type;
     }
 
     public void setType(ReservationType type) {
+        if (type == null) {
+            throw new IllegalArgumentException("Cannot create a reservation without a reservation type");
```
Comment (JortWillemsen) [inline src/main/java/nl/asd/reservation/domain/Reservation.java:67]: Change to: Reservation type cannot be null

File: src/main/java/nl/asd/reservation/domain/Reservation.java [hunk 5]
```diff
@@ -3,70 +3,86 @@
...
-        this.slots = slots;
-    }
-
     public WorkplaceId getWorkplace() {
         return workplace;
     }
 
     public void setWorkplace(WorkplaceId workplace) {
+        if (workplace == null) {
+            throw new IllegalArgumentException("Cannot create a reservation without an associated workplace");
```
Comment (JortWillemsen) [inline src/main/java/nl/asd/reservation/domain/Reservation.java:83]: Change to: associated workplace id cannot be null

File: src/main/java/nl/asd/reservation/domain/Timeslot.java [hunk 6]
```diff
@@ -30,4 +93,12 @@ public boolean equals(Object o) {
...
         return Objects.hash(from, to);
     }
+
+    @Override
+    public String toString() {
+        return "Timeslot{" +
+                "from=" + from +
+                ", to=" + to +
+                '}';
+    }
```
Comment (JortWillemsen) [inline src/main/java/nl/asd/reservation/domain/Timeslot.java:103]: Remove or rework

**Review by JortWillemsen — APPROVED** [review]

**Pull request merged**